pkg/util/slice/slice.go

```diff
@@ -0,0 +1,47 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package slice
+
+// SliceFilter retain elements for which the given predicate returns true
+func SliceFilter[T any](elements []T, predicate func(T) bool) []T {
```

Comment:
nitpick: usually, it is better to avoid repeating the package name in a function, i.e. `slice.Filter` instead of `slice.SliceFilter` as the package already defines its purpose  (or at least it should)

Reply:
ok